### PR TITLE
Add option to hide bottom bar when scrolling

### PR DESCRIPTION
**Pull request by scb261**
#3392, but I'm not sure if it really closes it. Some users including myself just don't like when bottom navbar is hidden, but in #3392 mentioned problem that the navbar won't show up in particular usecase, and the problem is still there if the option is set to default.

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -334,6 +341,7 @@ class MainActivity : BaseActivity<MainActivityBinding>() {
         if (to is RootController) {
             // Always show bottom nav again when returning to a RootController
             showBottomNav(visible = true, collapse = from !is RootController)
+            setBottomBarBehaviorOnScroll()
```
Comment (scb261) [inline app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt:344]: I'm not sure about this one. Right now it is called even when it isn't needed, but adding 
```
                onChange {
                    activity?.recreate()
                    true
                }
```
recreates whole activity and I'm not sure that it's right decision here.

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt:346]: Maybe just call it on the pref flow on each change?

Comment (scb261) [reply app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt:346]: I'm pretty new to Android development and Kotlin. How can I get current instance of `MainActivity` inside of `SettingsGeneralController`?

Comment (jobobby04) [reply app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt:346]: SettingsController should have the activity variable internally(as all controllers do)

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt:346]: I ended up making the change myself after merging this.

**Pull request merged**